tests/GslCore.Tests/testPromTermLen.fs

```diff
@@ -0,0 +1,85 @@
+﻿module testPromTermLen
+/// Test #promoterlen #terminatorlen work
+
+open NUnit.Framework
+open constants
+open commonTypes
+open pragmaTypes
+open Amyris.ErrorHandling
+
+/// location of test gslc_lib fixtures (depends on .net tooling version)
+let testLibDir1 = @"../../../../TestGslcLib"
+let testLibDir2 = @"../../../../../TestGslcLib"
+
+[<TestFixture>]
+type TestPromTermLen() = 
+    let emptyPragmas = PragmaCollection(Map.empty)
+    do
+        // initialize pragmas
+        pragmaTypes.finalizePragmas []
+
+    let testLibDir = 
+        if System.IO.Directory.Exists testLibDir1 then testLibDir1  
+            else testLibDir2
+
+    let same context expected actual =
+        if expected<>actual then 
+            failwithf "%s: expected= %d and actual=%d not equal" context expected actual
+
+    let checkOneGenome pragmas name promLen termLen termLenMRNA =
+        let gd = new RefGenome.GenomeDef(testLibDir,name)
+        gd.Load()
+        let part = DnaCreation.translateGenePrefix pragmas gd TERMINATOR
+        same "terminator length test" termLen ((part.right.x-part.left.x+1<OneOffset>)/1<OneOffset>) // +1 since ends are inclusive
+
+        let part = DnaCreation.translateGenePrefix pragmas gd PROMOTER
+        same "promoter length test" promLen ((part.right.x-part.left.x+1<OneOffset>)/1<OneOffset>)  // +1 since ends are inclusive
+
+        let mRNA = DnaCreation.translateGenePrefix pragmas gd MRNA
+        same "termmrna length test" termLenMRNA (((mRNA.right.x-1<OneOffset>)+1<OneOffset>)/1<OneOffset>) // Use 1 (rel to 3' end as the start of the terminator region
+
+    let testPragma name value refGenome expProm expTerm expTermMRNA =
+        match buildPragma name [value] with
+        | Ok (p,[]) ->
+            let map = [ p.name,p] |> Map.ofList |> PragmaCollection
+            checkOneGenome map refGenome expProm expTerm expTermMRNA 
+        | _ -> failwith "building promlen pragma"
+        
+    [<Test>]
+    member __.TestGenomesLoadable() =
+        let gd = new RefGenome.GenomeDef(testLibDir,"TestGenome")
+        gd.Load()
+        ()
+
+    [<Test>]
+    member __.TestPragmasExist() =
+        let checkPragmaExists name =
+            Assert.DoesNotThrow (fun () -> returnOrFail (buildPragma name ["250"]) |> ignore) |> ignore
```

Comment:
Matter of taste here, but asserting that an exception is not thrown is pretty much equivalent to running the action and allowing the test to fail if it does throw.  Compare to just

```
let checkPragmaExists name = buildPragma name ["250"] |> returnOrFail |> ignore
```